Assets/Scripts/UI/SwitchSelectMsg.cs

```diff
@@ -8,12 +8,24 @@ public class SwitchSelectMsg : MonoBehaviour {
 
 	private Controller2 controller2;
     public TextBoxManager textBoxManager;
-    // Use this for initialization
+
+	private checkOneTimeTrigger switchTriggeredCheck;
+	private checkOneTimeTrigger acceptTriggeredCheck;
+
+	private Renderer switchMessageRenderer;
+	private Renderer acceptMessageRenderer;
+
     //this is a tutotial script for level 1a
     void Start()
     {
         controller2 = GetComponent<Controller2>();
 		textBoxManager = GameObject.FindObjectOfType<TextBoxManager> ();
+
+		switchTriggeredCheck = switchMessage.GetComponent<checkOneTimeTrigger> ();
+		acceptTriggeredCheck = acceptMessage.GetComponent<checkOneTimeTrigger> ();
+		switchMessageRenderer = switchMessage.GetComponent<Renderer> ();
+		acceptMessageRenderer = acceptMessage.GetComponent<Renderer> ();
+
```

Comment:
it really depends Gowtham, cos if u only turn off the renderer, the whole game object is there and it will cause side effect to the scene.  Like imagine u turn off renderer of a collider, but the collider is still there.  Thou turning off renderer is lighter I believe.  and here since I still want the checkOneTimetrigger of the object to always be active, renderer is a better choice